spec/workers/polling_alert_check_worker_spec.rb

```diff
@@ -0,0 +1,88 @@
+RSpec.describe PollingAlertCheckWorker do
+  include SearchAlertListHelpers
+  include NotifyRequestHelpers
+
+  describe "#perform", caching: true do
+    def perform
+      described_class.new.perform("medical_safety_alert")
+    end
+
+    context "there are no alerts older than an hour" do
+      before { stub_medical_safety_alert_feed(content_id: SecureRandom.uuid, age: 30.minutes) }
+
+      it "should put 0/0 in the cache" do
+        expect(Rails.cache).to receive(:write).with("current_medical_safety_alerts", 0, expires_in: 15.minutes)
+        expect(Rails.cache).to receive(:write).with("delivered_medical_safety_alerts", 0, expires_in: 15.minutes)
+        perform
+      end
+    end
+
+    context "there are no alerts younger than 2 days" do
+      before { stub_medical_safety_alert_feed(content_id: SecureRandom.uuid, age: 3.days) }
+
+      it "should put 0/0 in the cache" do
+        expect(Rails.cache).to receive(:write).with("current_medical_safety_alerts", 0, expires_in: 15.minutes)
+        expect(Rails.cache).to receive(:write).with("delivered_medical_safety_alerts", 0, expires_in: 15.minutes)
+        perform
+      end
+    end
+
+    context "there is a valid alert with delivered emails" do
+      before do
+        content_id = SecureRandom.uuid
+        stub_medical_safety_alert_feed(content_id:, age: 2.hours)
+        create(:email, content_id:, notify_status: "delivered", status: "sent")
+      end
+
+      it "should put 1/1 in the cache" do
+        expect(Rails.cache).to receive(:write).with("current_medical_safety_alerts", 1, expires_in: 15.minutes)
+        expect(Rails.cache).to receive(:write).with("delivered_medical_safety_alerts", 1, expires_in: 15.minutes)
+        perform
+      end
+    end
+
+    context "there is a valid alert with undelivered emails" do
+      before do
+        content_id = SecureRandom.uuid
+        stub_medical_safety_alert_feed(content_id:, age: 2.hours)
+        email = create(:email, content_id:, notify_status: nil, status: "sent")
+        stub_notify_email_delivery_status_delivered_empty(email.id)
+      end
+
+      it "should put 1/0 in the cache" do
+        expect(Rails.cache).to receive(:write).with("current_medical_safety_alerts", 1, expires_in: 15.minutes)
+        expect(Rails.cache).to receive(:write).with("delivered_medical_safety_alerts", 0, expires_in: 15.minutes)
+        perform
+      end
+    end
+
+    context "there is a valid alert with apparently undelivered emails but they actually were delivered" do
```

Comment:
-> "there are no alerts as polling confirmed they were actually delivered"

Reply:
I've renamed all of these to make the wording clear that it's "email alerts" we're concerned about in the context (ie the thing we're watching), not a generated alert (ie what Prometheus does when the two metrics don't match)